### PR TITLE
Assets on s3

### PR DESCRIPTION
**Pull request by adgad**
/cc @matthew-andrews @commuterjoy @richard-still-ft 

Task to deploy static assets to Amazon S3

### CODE REVIEW

**Comment by matthew-andrews** [comment]
can we keep the same api and have the `deploy-hashed-assets` method deploy to both (initially optionally to S3 — warning but not failing if no API key set)?

**Comment by adgad** [comment]
@matthew-andrews like so?

File: bin/next-build-tools.js [hunk 1]
```diff
@@ -97,6 +98,7 @@ program
 		.description('deploys ./hashed-assets/ to <app-name> on GitHub')
 		.action(function() {
 			deployHashedAssets().catch(exit);
+			deployHashedAssetsToS3().catch(exit);
```
Comment (matthew-andrews) [inline bin/next-build-tools.js]:
```js
Promise.all([
  deployHashedAssets,
  deployHashedAssetsToS3
]).catch(exit);
```

File: package.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "test": "./bin/next-build-tools.js verify"
   },
   "dependencies": {
+    "aws-sdk": "^2.1.19",
```
Comment (matthew-andrews) [inline package.json]: urgh that dep is HUGE

**Comment by adgad** [comment]
hmm...let me rewrite using the REST api.

**Comment by adgad** [comment]
Okay I'm giving up on trying to get the REST api working for now :( Will get this through/working and can optimise later if needed.

**Comment by matthew-andrews** [comment]
:+1:

**Pull request merged**